pkg/app/server/apikeyverifier/verifier.go

```diff
@@ -67,6 +68,11 @@ func (v *Verifier) Verify(ctx context.Context, key string) (*model.APIKey, error
 		return nil, fmt.Errorf("unable to find API key %s from datastore, %w", keyID, err)
 	}
 
+	// update the time API key was last used
+	if err := v.apiKeyStore.UpdateLastUsedAt(ctx, keyID, apiKey.ProjectId); err != nil {
```

Comment:
if only check it here, the UpdateLastUsedAt only be triggered on case of apiKeyCache missed, isn't it?